src/test/kotlin/org/opensearch/replication/task/index/NoOpClient.kt

```diff
@@ -110,7 +109,7 @@ open class NoOpClient(testName :String) : NoOpNodeClient(testName) {
             val indexToSettings = HashMap<String, Settings>()
             indexToSettings[IndexReplicationTaskTests.followerIndex] =  desiredSettingsBuilder.build()
 
-            val settingsMap = ImmutableOpenMap.builder<String, Settings>().putAll(indexToSettings).build()
+            val settingsMap = indexToSettings.toMutableMap()
```

Comment:
should be immutable map

Reply:
We dont want to be using Immutable as map as that implementation is being remove.

Reply:
ImmutableOpenMap is being removed. So you can use any other immutable implementation like `Collections.unmodifiableMap`